lib/install/validate/validator.go

```diff
@@ -482,3 +512,68 @@ func (v *Validator) AddDeprecatedFields(ctx context.Context, conf *config.Virtua
 
 	return &dconfig
 }
+
+// helper to grab HostSystem references out of the Session cluster
+func (v *Validator) getHostsInCluster(ctx context.Context) ([]*object.HostSystem, error) {
+	computeHosts, err := v.Session.Cluster.Hosts(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// computeHosts shouldn't be zero anymore; if it still is, there's a problem
+	if len(computeHosts) == 0 {
+		return nil, errors.New("Couldn't get list of hosts connected to target cluster")
+	}
+	return computeHosts, nil
+}
+
+// checks if a given datastore ds is writeable by compute resource specified on cmd line --compute-resource
+func (v *Validator) isTargetDatastoreWriteable(ctx context.Context, ds *object.Datastore) bool {
+	defer trace.End(trace.Begin(ds.String()))
+
+	// get compute hosts if we haven't run this method on this object before
+	if len(v.computeHosts) == 0 {
+		var err error
+		v.computeHosts, err = v.getHostsInCluster(ctx)
+		if err != nil {
+			v.NoteIssue(err)
+		}
+	}
+
+	// now get hosts connected to ds
+	attachedHosts, err := ds.AttachedClusterHosts(ctx, v.Session.Cluster)
```

Comment:
The AttachedClusterHosts method will check if this datastore is connected to cluster hosts, which does not provide value for your current function. 
ds.AttachedHosts is good enough.